kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/impl/ContainerOrchestrationServiceImpl.java

```diff
@@ -684,45 +683,37 @@ private HostConfig containerRuntimeConfigurationHandler(ContainerConfiguration c
         return configuration;
     }
 
-    private HostConfig containerPortManagementHandler(ContainerConfiguration containerDescription,
-            HostConfig commandBuilder) {
+    private HostConfig containerPortManagementHandler(ContainerConfiguration containerConfiguration,
+            HostConfig hostConfig, CreateContainerCmd commandBuilder) {
 
-        if (containerDescription.getContainerPorts() != null && !containerDescription.getContainerPorts().isEmpty()) {
-            List<ExposedPort> exposedPortsList = new LinkedList<>();
-            Ports portbindings = new Ports();
+        List<ExposedPort> exposedPorts = new LinkedList<>();
+        Ports portbindings = new Ports();
 
-            for (org.eclipse.kura.container.orchestration.ContainerPort port : containerDescription
-                    .getContainerPorts()) {
+        if (containerConfiguration.getContainerPorts() != null
+                && !containerConfiguration.getContainerPorts().isEmpty()) {
 
-                InternetProtocol ipPro;
+            List<org.eclipse.kura.container.orchestration.ContainerPort> containerPorts = containerConfiguration
+                    .getContainerPorts();
 
-                switch (port.getInternetProtocol()) {
-                case UDP:
-                    ipPro = InternetProtocol.UDP;
-                    break;
-                case SCTP:
-                    ipPro = InternetProtocol.SCTP;
-                    break;
-                default:
-                    ipPro = InternetProtocol.TCP;
-                    break;
-                }
-
-                ExposedPort tempExposedPort = new ExposedPort(port.getInternalPort(), ipPro);
-                exposedPortsList.add(tempExposedPort);
-                portbindings.bind(tempExposedPort, Binding.bindPort(port.getExternalPort()));
-            }
+            for (org.eclipse.kura.container.orchestration.ContainerPort port : containerPorts) {
 
-            commandBuilder.withPortBindings(portbindings);
+                InternetProtocol ipProtocol = InternetProtocol.parse(port.getInternetProtocol().toString());
```

Comment:
This changes the behaviour slightly, in the previous implementation `TCP` would be used as a default if port.getInternetProtocol() returns null, now the code will throw a NullPointerException in the same case.

Reply:
port.getInternetProtocol() cannot be null because is set in ContainerPort constructors and cannot be something different from one item of org.eclipse.kura.container.orchestration.PortInternetProtocol enum

Reply:
nothing stops the user from passing null as third argument here https://github.com/eclipse/kura/blob/6629ddf759d88af03b1da88bc27a3fc82253ce38/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ContainerPort.java#L30